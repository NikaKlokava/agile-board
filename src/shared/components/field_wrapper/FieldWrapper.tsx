import { ReactNode } from "react";
import classes from "classnames";
import cl from "./field_wrapper.module.css";

type Props = {
  children: ReactNode;
  fieldName: string;
  clName?: "style_container";
};

export const FieldWrapper = ({ fieldName, children, clName }: Props) => {
  return (
    <div className={classes(cl.container, clName && cl[clName])}>
      <p className={cl.title}>{fieldName}</p>
      {children}
    </div>
  );
};
