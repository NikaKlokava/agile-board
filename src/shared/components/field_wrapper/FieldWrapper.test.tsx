import renderer from "react-test-renderer";
import { FieldWrapper } from "./FieldWrapper";

describe("Test the FieldWrapper component", () => {
  test("The FieldWrapper renders correctly", () => {
    const fieldWrapperSnap = renderer
      .create(<FieldWrapper children={undefined} fieldName={"TestName"} />)
      .toJSON();
    expect(fieldWrapperSnap).toMatchSnapshot();
  });
});
