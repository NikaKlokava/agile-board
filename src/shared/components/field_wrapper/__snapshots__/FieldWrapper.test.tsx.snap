// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the FieldWrapper component The FieldWrapper renders correctly 1`] = `
<div
  className="container"
>
  <p
    className="title"
  >
    TestName
  </p>
</div>
`;

exports[`Test the Input component The Input renders correctly 1`] = `
<div
  className="container"
>
  <p
    className="title"
  >
    TestName
  </p>
</div>
`;
